eth/handler.go

```diff
@@ -399,3 +410,25 @@ func (h *handler) txBroadcastLoop() {
 		}
 	}
 }
+
+// missingBodyLoop listens to the MissingBody event in Slice and calls the blockAnnounces.
+func (h *handler) missingBodyLoop() {
+	defer h.wg.Done()
+	for {
+		select {
+		case header := <-h.missingBodyCh:
+			// Check if any of the peers have the body
+			for _, peer := range h.peers.peers {
+				if peer.KnownBlock(header.Hash()) {
```

Comment:
We shouldn't filter on `KnownBlock`. If we missed a body announcement (e.g. if node was offline), then we won't have the body AND the block won't be registered as a known block for any of our peers. This is the exact scenario where we need to find a peer to ask for the body, but the ask won't happen, because none of our peers have this listed as a known block.

Reply:
Not using KnownBlock anymore